Template Support (XML, JSON)

SirRamEsq opened the pull request:
This will resolve issue #74 (Tiled Template Support),

This has been tested on Godot 3.0.4. XML and JSON TileMaps and Templates have both been tested and should work.

Templates with and without GIDs (tiles) have also been tested to work.

If a template is changed, the maps are not automatically re-imported.

vnen commented:
Cool, thanks for the contribution. Can you please [squash the commits together](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase)? I'll try to review this tomorrow.

SirRamEsq commented:
No problem,
I just pushed the squashed commit, so we should be all set.

vnen commented:
Sorry, I still couldn't find time to look into this yet, but it's in my todo list.

Just a comment: why did you change the functions to be static? They don't rely on any object state, but there's not much reason to make them static (unless we make them all static and avoid the object instantiation, but that would be better in another discussion).

SirRamEsq commented:
Sorry it took awhile to reply, I was on a trip in the Ukraine last week. 

I call `parse_object()` statically to load a template file. The other functions were marked as static because they are called by `parse_object()` and need to also be called statically

Like you said, none of those functions rely on state, so it may not be a bad idea to just have them all be static. For the purposes of this pull request, I only changed the functions I needed to be static

SirRamEsq commented:
Updated pull request today.

When opening the project in Windows, I found a bug with my import that failed to find the template file. Fixed the bug, squashed the commits together, and forced push to my branch.
